src/Layers/xrRender/DetailManager_CACHE.cpp

```diff
@@ -197,17 +201,20 @@ void CDetailManager::cache_Update(int v_x, int v_z, Fvector& view)
             float D = view.distance_to_sqr(C);
 
             // Select
-            for (int j = 0; j < dm_max_decompress; ++j)
+            if (D < bestDistances[maxIdx])
             {
-                if (D < bestDistances[j])
-                {
-                    bestDistances[j] = D;
-                    bestIndexes[j] = i;
-                }
+                bestDistances[maxIdx] = D;
+                bestIndexes[maxIdx] = i;
+
+                const auto maxIt = std::max_element(bestDistancesBegin, bestDistancesEnd);
+                maxIdx = std::distance(bestDistancesBegin, maxIt);
             }
         }
 
-        for (int i = 0; i < dm_max_decompress; ++i)
+        // We sort indexes and process them in reverse order to erase correctly from cache_task
```

Comment:
```suggestion
        // Because indexes become invalid after an erase, sort them and process in the reverse order to erase everything correctly
```